src/Controls/src/Core/Handlers/Shell/ShellHandler.Windows.cs

```diff
@@ -123,5 +163,81 @@ public static void MapFlyoutItems(ShellHandler handler, Shell view)
 		{
 			handler.PlatformView.UpdateMenuItemSource();
 		}
+
+		void UpdateFlyoutHeaderBehavior(Shell view)
+		{
+			var flyoutHeader = (ShellHeaderView)PlatformView.PaneCustomContent;
+
+			if (view.FlyoutHeaderBehavior == FlyoutHeaderBehavior.Default ||
+				view.FlyoutHeaderBehavior == FlyoutHeaderBehavior.Fixed)
+			{
+				var defaultHeight = headerHeight;
+				var defaultTranslateY = headerOffset;
+
+				UpdateFlyoutHeaderTransformation(flyoutHeader, defaultHeight, defaultTranslateY);
+				return;
+			}
+
+			var menuItemsScrollViewer = _scrollViewer;
+
+			if (menuItemsScrollViewer != null)
+			{
+				double? topAreaHeight = null;
+
+				menuItemsScrollViewer.ViewChanged += (sender, args) =>
```

Comment:
Should this unsubscribe? I think you'll just keep betting subscriptions here if the user toggles `FlyoutHeaderBehavior`

Reply:
Yes, we must unsubscribe before subscribe. Updated.